sisl/geometry.py

```diff
@@ -3810,6 +4124,38 @@ def __call__(self, parser, ns, value, option_string=None):
                            action=PeriodTileZ,
                            help='Tiles the geometry along the third cell vector.')
 
+        # Sort
+        class Sort(argparse.Action):
+            def __call__(self, parser, ns, values, option_string=None):
+                # call geometry.sort(...) using appropriate keywords (and ordered dict)
+                kwargs = OrderedDict()
+                opts = values[0].split(';')
+                for i, opt in enumerate(opts):
+                    # Split for equal
+                    opt = opt.split('=', 1)
+                    if len(opt) > 1:
+                        opt, val = opt
+                    else:
+                        opt = opt[0]
+                        val = "True"
+                    if val.lower() in ['t', 'true']:
+                        val = True
+                    elif val.lower() in ['f', 'false']:
+                        val = False
+                    elif opt in ['atol']:
+                        # float values
+                        val = float(val)
+                    else:
+                        # it must be a range/tuple
+                        val = lstranges(strmap(int, val))
+                    # parse `val` to appropriate values
+                    # we always add integers to allow users to use the same keywords on commandline
+                    kwargs[opt.strip() + str(i)] = val
+                ns._geometry = ns._geometry.sort(**kwargs)
+        p.add_argument(*opts('--sort'), nargs=1, metavar='SORT',
+                       action=Sort,
+                       help='Semi-colon separated options for sort [axis=0;descend;lattice=(1, 2)].')
```

Comment:
Maybe stress that the CLI argument needs to be given with quotation marks, as in `--sort "axis=0;descend;lattice=(1, 2)"`.

These calls are invalid: `--sort axis=0;descend;lattice=(1,2)` and `--sort [axis=0;descend;lattice=(1, 2)]` (and the error message not too helpful to localize the mistake).

Reply:
thanks